src/StorageApi/ConfigurationCorrector.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProjectRestore\StorageApi;
+
+use Psr\Log\LoggerInterface;
+use stdClass;
+
+class ConfigurationCorrector
+{
+    private StackSpecificComponentIdTranslator $componentIdTranslator;
+
+    public function __construct(
+        string $destinationApiUrl,
+        LoggerInterface $logger
+    ) {
+        $destinationStack = self::getStackFromUrl($destinationApiUrl);
+
+        $this->componentIdTranslator = new StackSpecificComponentIdTranslator(
+            $destinationStack,
+            $logger
+        );
+    }
+
+    public function correct(string $componentId, stdClass $configuration): stdClass
+    {
+        if ($componentId === 'keboola.orchestrator') {
+            $configuration = $this->correctOrchestratorConfiguration($configuration);
+        }
+
+        $configuration = $this->removeOauthAuthorization($configuration);
+
+        return $configuration;
+    }
+
+    private function correctOrchestratorConfiguration(stdClass $configuration): stdClass
+    {
+        foreach ($configuration->tasks ?? [] as $task) {
+            $componentId = &$task->task->componentId;
```

Comment:
přes ty reference je to takový dost nepřehledný, protože vytvoříš proměnou componentId který pak nikde nepoužiješ

Reply:
To máš pravdu. Dal jsem pryč.